diracx-db/src/diracx/db/sql/utils.py

```diff
@@ -212,10 +264,62 @@ async def ping(self):
             raise SQLDBUnavailable("Cannot ping the DB") from e
 
 
-def apply_search_filters(table, stmt, search):
-    # Apply any filters
+def find_time_resolution(value):
+    if isinstance(value, datetime):
+        return None, value
+    if match := re.fullmatch(
+        r"\d{4}(-\d{2}(-\d{2}(([ T])\d{2}(:\d{2}(:\d{2}(\.\d{6}Z?)?)?)?)?)?)?", value
+    ):
+        if match.group(6):
+            precision, pattern = "SECOND", r"\1-\2-\3 \4:\5:\6"
+        elif match.group(5):
+            precision, pattern = "MINUTE", r"\1-\2-\3 \4:\5"
+        elif match.group(3):
+            precision, pattern = "HOUR", r"\1-\2-\3 \4"
+        elif match.group(2):
+            precision, pattern = "DAY", r"\1-\2-\3"
+        elif match.group(1):
+            precision, pattern = "MONTH", r"\1-\2"
+        else:
+            precision, pattern = "YEAR", r"\1"
+        return (
+            precision,
+            re.sub(
+                r"^(\d{4})-?(\d{2})?-?(\d{2})?[ T]?(\d{2})?:?(\d{2})?:?(\d{2})?\.?(\d{6})?Z?$",
+                pattern,
+                value,
+            ),
+        )
+
+    raise InvalidQueryError(f"Cannot parse {value=}")
+
+
+def apply_search_filters(column_mapping, stmt, search):
```

Comment:
as a general consideration, it's probably better to give the `table` instead of the column mapping. The reason is that if SQLAlchemy changes the internals (and it certainly wouldn't be the first time), we would only need to fix it here, as opposed to everywhere we call this method. It's also more readable

Reply:
This is something I plan to refactor to make it so we can handle joins (which we already need) so I propose to do it in my next PR?

Reply:
Works with me !